File: backend/app/schemas/sales.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class SaleItemCreate(BaseModel):
    product_id: int
    category_id: int
    quantity: int = Field(gt=0)
    unit_price: float = Field(gt=0)
    discount: float = Field(default=0, ge=0)
    tax: float = Field(default=0, ge=0)


class SaleCreate(BaseModel):
    customer_id: int
    payment_method: str
    notes: Optional[str] = None
    discount: float = Field(default=0, ge=0)
    tax: float = Field(default=0, ge=0)
    items: list[SaleItemCreate]


class SaleItemResponse(BaseModel):
    id: int
    product_id: int
    category_id: int
    quantity: int
    unit_price: float
    discount: float
    tax: float
    total: float

    class Config:
        from_attributes = True


class SaleResponse(BaseModel):
    id: int
    invoice_number: str
    customer_id: int
    customer_name: str
    sale_date: datetime
    payment_method: str
    notes: Optional[str]
    subtotal: float
    discount: float
    tax: float
    total_amount: float
    status: str
    created_by: int
    item_count: Optional[int] = None

    class Config:
        from_attributes = True

File: frontend/src/types/sales.ts
export interface SaleItem {
    product_id: number;
    category_id: number;
    quantity: number;
    unit_price: number;
    discount: number;
    tax: number;
    total?: number;
}

export interface Sale {
  id?: number;
  invoice_number?: string;
  customer_id: number;
  customer_name: string;
  sale_date?: string;
  payment_method: string;
  notes?: string;
  subtotal?: number;
  discount?: number;
  tax?: number;
  total_amount?: number;
  status?: string;
  created_by?: number;
  items?: SaleItem[];
}

export interface DashboardSummary {
    total_sales: number;
    total_revenue: number;
    total_orders: number;
    average_order_value: number;
}

export interface Notification {
    id: number;
    title: string;
    message: string;
    status: string;
    created_at: string;
}